myia/operations/op_zeros_like.py

```diff
@@ -59,6 +60,13 @@ def _array_zero(xs):
     return distribute(myia_to_array(scalar_zero, typeof(xs)), shape(xs))
 
 
+@_leaf_zeros_like.register(AbstractRandomState)
+@core
+def _rng_zero(x):
+    """Return random_initialize with seed 0 by default."""
+    return random_initialize(0)
```

Comment:
That's not really a zero, though, is it?

Insofar that it is used for the purpose of gradients, `zeros_like` does not necessarily have to return the same type as its input (e.g. on functions it returns newenv). It has to match the signature of `sensitivity_transform`, which I think is somewhere in `abstract.utils`. So it *could* simply return 0, I think, provided that `sensitivity_transform` maps the `AbstractRandomState` type to `AbstractScalar`.

Ideally I think it would return `AbstractBottom` which has the interesting property that it produces an error if it's used, but I think that would mess everything up right now.

Reply:
Indeed, it's not really a zero.

I updated zeros_like and sensitivity_transform so that zeros_like(rstate) returns zero scalar.